### ISSUE TITLE
YAML Reserved Characters (@ named matcher throws parse error)

### ISSUE THREAD

**Issue by RenatoXSR** [lucaslorentz/caddy-docker-proxy#412]
I am trying to define some matchers in my docker compose file, but I get an illegal char error.

I verified that, actually, beginning a token with @ is [reserved in YAML specs](https://yaml.org/spec/1.2.2/#rule-c-directive), so we may need to find an alternate character to use in the labels, or maybe create a 'named matcher' directive and alter it during isMatcher check in  [caddyfile/merge.go`](https://github.com/lucaslorentz/caddy-docker-proxy/blob/master/caddyfile/merge.go#L40) or somewhere similar when merging or comparing blocks, after the labels have been parsed.

Are there any suggestions of which way is better, and what char we could use?

Reserved characters: `-`, `?`, `:`, `,`, `[`, `]`, `{`, `}`, `#`, `&`, `*`, `!`, `|`, `>`, `'`, `"`, `%`, `@` and `.

**Comment by lucaslorentz** [comment]
Hey Renato,

Our README indeed have invalid yaml as examples, like:
```
caddy.@match.path: /test
caddy.reverse_proxy: @match localhost:6001
```

That's because I wrote the examples without an yaml validator 😄  But basically all values we add after `:` are yaml strings, whenever they start with a special yaml character you can wrap it with `"`.

Example:
```
caddy.@match.path: /test
caddy.reverse_proxy: "@match localhost:6001"
```

I hope this solve your issue. If that's not the problem, can you please share a yaml example to make it more clear?

**Comment by RenatoXSR** [comment]
@lucaslorentz thank you very much for the immediate reply ;-)

That it exactly my case. 

I assumed the error was the label `caddy.@matcher` and not the YAML string after the `:`, and it is difficult to track `Caddyfile` validation errors, because `caddy-docker-proxy` removes the entire invalid block, without specifying which line has the error (there are a few closed issues explaining why it is so and cannot be more specific).

I'll make a PR complementing the README example.

Would it be ok if I also included a similar example, but regarding JSON respond blocks?

As the YAML string after the `:` is parsed by `caddy-docker-proxy` before writing the blocks to the generated `Caddyfile`, I couldn't just use a label like `caddy.respond: "{\"key\":\"value\"}"`, so I escaped all the `{`, `}`, `:` and `"` as `\uXXXX` codes (respectively, `\u007b`, `\u007d`, `\u003a` and `\u0022`).

Cheers!
